query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -249,6 +273,40 @@ abstract class LeafExecPlan extends ExecPlan {
   final def children: Seq[ExecPlan] = Nil
 }
 
+sealed trait FuncArgs{
+  def getResult (implicit sched: Scheduler, timeout: FiniteDuration) : Observable[ScalarVector]
+}
+case class ExecPlanFuncArgs(execPlan: ExecPlan) extends FuncArgs {
+  override def getResult(implicit sched: Scheduler, timeout: FiniteDuration): Observable[ScalarVector] = {
+    Observable.now(execPlan)
+      .mapAsync(Runtime.getRuntime.availableProcessors()) { plan =>
```

Comment:
Instead of using `Observable.now(...).mapAsync` it is easier just to create the task from dispatch, then use `Observable.fromTask(...)`